tardis/adapters/sites/slurm.py

```diff
@@ -58,8 +58,14 @@ async def squeue(
         raise
 
     else:
+        # Not all lines returned in slurm_status.stdout belong to the actual squeue output
+        # Only keep lines containing the delimiter
+        cleaned_stdout = []
+        for row in slurm_status.stdout.splitlines():
+            if "|" in row:
+                cleaned_stdout.append(row)
         for row in csv_parser(
```

Comment:
It might be better to directly use [`csv.DictReader`](https://docs.python.org/3/library/csv.html#csv.DictReader) here. This would avoid the useless roundtrip of creating an iterable of lines, joining that to newline separated lines, and then creating an iterable of lines from it again.  
Not sure how straightforward that is nor whether I'm overlooking something, though.

I'm fine either way, please follow your preference.

Reply:
I think I should not touch this if there is no clear request from you side. So I leave it for now.

Reply:
In addition, it would be nice to have a unittest for that issue as well. It should go to the following class https://github.com/MatterMiners/tardis/blob/4bc109d32ecd344ca1c13c5f79a16cc4ec4bae11/tests/utilities_t/test_utils.py#L90

Reply:
I thought as well that the csv_parser could be modified to skip lines which don't contain the delimiter. But I wasn't sure if this is always intended or expected. Should we add another parameter skiplineswithoutdelimiter with some default to make it adjustable?

Reply:
Should the unittest be a separate test, which would duplicate some of the existing tests' code, or just an additional nonsense line in the input of one of the existing tests?

Reply:
> Should the unittest be a separate test, which would duplicate some of the existing tests' code, or just an additional nonsense line in the input of one of the existing tests?

Should be a separate member function of that class.

Reply:
> I thought as well that the csv_parser could be modified to skip lines which don't contain the delimiter. But I wasn't sure if this is always intended or expected. Should we add another parameter skiplineswithoutdelimiter with some default to make it adjustable?

I think we have no use-case, where the delimiter is expected to be missing. To be save, I would suggest to require `len(fieldnames)>1` in addtion, so the delimiter is expected to exists. There is still the possibility to run in this issue, if the delimiter is a space. However, I think this something that is hard to avoid and in addition, our site adapters currently use either `\t` or `|` as delimiter.